docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
 
-### Value proposition
+## 2.0 Setting up the project in your computer
+This sections gives an overview on how to set up the project in your computer.
 
-{Describe the value proposition: what problem does it solve?}
+**Prerequisites:**
+* JDK 11
+* IntelliJ IDE
 
-## User Stories
+<div markdown="span" class="alert alert-warning">
 
-|Version| As a ... | I want to ... | So that I can ...|
+⚠️ **Caution:**
+Follow the steps in the following guide precisely. Things will not work out if you deviate in some steps.
+</div>
+
+First, **fork** this repo, and **clone** the fork into your computer.
+
+If you plan to use Intellij IDEA (highly recommended):
+1. **Configure the JDK**: Follow the guide [_[se-edu/guides] IDEA: Configuring the JDK_](https://se-education.org/guides/tutorials/intellijJdk.html) to to ensure Intellij is configured to use **JDK 11**.
+1. **Import the project as a Gradle project**: Follow the guide [_[se-edu/guides] IDEA: Importing a Gradle project_](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into IDEA.
+:exclamation: Note: Importing a Gradle project is slightly different from importing a normal Java project.
+1. **Verify the setup**:
+   1. Run the `seedu.trippie.Trippie` and try a few commands.
+   2. [Run the tests](TestingGuide.md) to ensure they all pass. 
+
+## 3.0 Design
+
+This section explains the high-level design of the application. Given below is an overview of each component and a more detailed explanation of the architecture. Trippie is the main class which manages the initialization of the relevant classes and their execution.
+
+
+### 3.1 Architecture (Felix)
+
+<kbd>
+
+![UML Diagram Main](https://i.imgur.com/V3vUpoo.jpg)
+<center><i>Figure 1: Overall Architecture</i></center>
+
+</kbd>
+
+The Architecture Diagram given above explains the high-level design of Trippie. Shown below is a diagram of Trippie's high-level structure.
+
+The main class of`Trippie` is called [`Trippie.java`](https://github.com/AY2021S1-CS2113T-W11-2/tp/blob/master/src/main/java/seedu/trippie/Trippie.java?) It is responsible for:
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+
+The rest of the App consists of three components.
+- **UI** : The UI of the App.
+- **Parser** : Parses user input and its information to respective commands.
+- **Storage** : Reads data from, and write data to, the text file.
+- **Command** : Executes each command respectively.
+
+The Sequence diagram below shows how the program flows across the different classes when parsing or executing user commands.
+
+![UML Sequence Diagram](https://i.imgur.com/hHq7ltY.png)
```

Comment:
The method call for functions requires "()" at the end. 
If I am not wrong, the command should end after the execute activation bar. With a new command, it will reinitialize the command object. Putting a X there would be good.

![image](https://user-images.githubusercontent.com/49357313/97659760-c3ff8f00-1aaa-11eb-8c52-ce781234aa2e.png)